engine/src/gldrv/gl_light_state.cpp

```diff
@@ -464,6 +464,61 @@ void gfx_light::ResetProperties(const enum LIGHT_TARGET light_targ, const GFXCol
     }
 }
 
+void gfx_light::ResetProperties(const enum LIGHT_TARGET light_target, const Vector& vector)
```

Comment:
So, following your comment from above about using vector for position and color for well color, I assume the right way would be to split this function to two.
For vector, only support position. For color support all others. Except attenuation seems to use position for something using target. 
Enough, I'm  going to assume you've done your due diligence.